packages/rabbitmq/src/utils.ts

```diff
@@ -0,0 +1,128 @@
+import { CSL, Cardano, cslToCore } from '@cardano-sdk/core';
+import { OutsideOfValidityInterval } from '@cardano-ogmios/schema';
+import { fromSerializableObject, toSerializableObject } from '@cardano-sdk/util';
+
+export interface DeserializedError {
+  innerError?: {
+    name: string;
+    message: string;
+  };
+  name: string;
+  message: string;
+}
+
+export interface SerializedError {
+  value: {
+    innerError?: {
+      name: string;
+      message: string;
+      stack?: string;
+    };
+    name: string;
+    message: string;
+    stack?: string;
+  };
+}
+
+export const TX_SUBMISSION_QUEUE = 'cardano-tx-submit';
+
+/**
+ * Enriches an error with the right prototype
+ *
+ * @param err the error to enrich
+ */
+const addPrototypeToError = (err: DeserializedError) => {
+  const { name, innerError } = err;
+  let prototype: object | null = Error.prototype;
+
+  if (name in Cardano.TxSubmissionErrors)
+    prototype = Cardano.TxSubmissionErrors[name as keyof typeof Cardano.TxSubmissionErrors].prototype;
+
+  Object.setPrototypeOf(err, prototype);
+
+  if (innerError) addPrototypeToError(innerError);
+};
+
+/**
+ * Deserializes a serialized error
+ *
+ * @param serializedError the error to deserialize
+ */
+export const deserializeError = (serializedError: SerializedError): unknown => {
+  const ret = fromSerializableObject(serializedError) as DeserializedError;
+
+  addPrototypeToError(ret);
+
+  return ret;
+};
+
+/**
+ * Serializes an error and checks if it is retriable
+ *
+ * @param err the error to serialize
+ */
+export const serializeError = (err: unknown) => {
+  let isRetriable = false;
+  let serializedError: SerializedError | null = null;
+
+  const lastChance = () => {
+    try {
+      serializedError = { value: { message: JSON.stringify(err), name: 'Error' } };
+    } catch {
+      serializedError = { value: { message: 'Unknown', name: 'Error' } };
+    }
+  };
+
+  try {
+    serializedError = toSerializableObject(err) as SerializedError;
+  } catch {
+    lastChance();
```

Comment:
That might be an overkill as it should never throw.

Either way you should probably return a UnknownTxSubmissionError in `lastChance` (which is one of the options in `Cardano.TxSubmissionError`) here, so we can assure that any error returned by this service would be a `TxSubmissionError`